### ISSUE TITLE
Specify individual resource requests for the synchronizer

### ISSUE THREAD

**Issue by yfei1** [googleforgames/open-match#1023]
One of the things that I noticed while benchmarking Open Match is the synchronizer as the single point of failure may requires more CPU resources other than the other components. Right now, we unified the resource requirements for all pods using the same value which may lead to CPU throttle on the synchronizer. We should split up the number when we finished the benchmark story. @Laremere could you also benchmark the CPU usage of your synchronizer implementation comparing to the old one when you check the code in?

**Comment by sawagh** [comment]
I propose moving this to v0.10 given this does not actually impact integration directly - unless this turns out to be a blocker for scaling.

**Comment by yfei1** [comment]
Before the evaluator id change #1082, memory is the bottleneck for both the synchronizer and the evaluator and I often see pods get OOM killed under load. Thanks to that change it seems like the memory request is no longer a bottleneck for both services. Here is what I am able to get with `kubectl top pod -n open-match` when running the scale test with batteroyale scenario and 4.6K CreateTicketQPS. Closing this as this issue has been fixed in v0.9.

```
NAME                                                        CPU(cores)   MEMORY(bytes)   
om-backend-669f88bd7c-4bzg7                                 820m         8Mi             
om-backend-669f88bd7c-qg826                                 623m         9Mi             
om-backend-669f88bd7c-qmhgx                                 643m         9Mi             
om-evaluator-794d7b78fd-7kwlv                               11m          4Mi             
om-evaluator-794d7b78fd-9g2bl                               12m          4Mi             
om-evaluator-794d7b78fd-9j2rz                               13m          4Mi             
om-frontend-74ddf84b46-6rk4j                                906m         8Mi             
om-frontend-74ddf84b46-dlqmd                                904m         8Mi             
om-frontend-74ddf84b46-fkb4t                                929m         8Mi             
om-frontend-74ddf84b46-k9x2z                                872m         9Mi             
om-frontend-74ddf84b46-ldssk                                902m         8Mi             
om-frontend-74ddf84b46-sz2sr                                955m         8Mi             
om-frontend-74ddf84b46-wjrc2                                952m         8Mi             
om-function-6bb4b468d4-7nhs2                                57m          5Mi             
om-function-6bb4b468d4-dxwl2                                58m          6Mi             
om-function-6bb4b468d4-lk78x                                55m          6Mi             
om-query-556dc456c9-8z5sd                                   137m         8Mi             
om-query-556dc456c9-p5zqf                                   167m         8Mi             
om-query-556dc456c9-qgwl9                                   160m         9Mi             
om-redis-master-0                                           1006m        33Mi            
om-redis-slave-0                                            253m         15Mi            
om-redis-slave-1                                            212m         17Mi            
om-redis-slave-2                                            182m         15Mi            
om-scale-backend-6668fcc566-j86ln                           982m         9Mi             
om-scale-frontend-747c989cd9-4mhwp                          266m         7Mi             
om-scale-frontend-747c989cd9-bzncs                          298m         7Mi             
om-scale-frontend-747c989cd9-cwmqz                          281m         7Mi             
om-scale-frontend-747c989cd9-gtxfw                          276m         7Mi             
om-scale-frontend-747c989cd9-h9vnd                          265m         7Mi             
om-scale-frontend-747c989cd9-nvcs7                          235m         7Mi             
om-scale-frontend-747c989cd9-p7krx                          284m         7Mi             
om-scale-frontend-747c989cd9-pk5xn                          305m         7Mi             
om-scale-frontend-747c989cd9-psqkl                          278m         7Mi             
om-scale-frontend-747c989cd9-zsm7j                          261m         8Mi             
om-swaggerui-8b9d84c95-h26hb                                1m           5Mi             
om-synchronizer-596cfdc4ff-r96tk                            121m         8Mi             
open-match-grafana-7f6744756b-xc9nm                         6m           78Mi            
open-match-prometheus-alertmanager-5cd5bbd7f4-mfcx2         2m           10Mi            
open-match-prometheus-kube-state-metrics-6b4df4584b-t54jx   4m           11Mi            
open-match-prometheus-node-exporter-2zhbc                   0m           8Mi             
open-match-prometheus-node-exporter-5jcv8                   0m           7Mi             
open-match-prometheus-node-exporter-7zr5t                   0m           9Mi             
open-match-prometheus-node-exporter-glt7g                   1m           9Mi             
open-match-prometheus-node-exporter-gms8l                   0m           8Mi             
open-match-prometheus-node-exporter-pxsdh                   0m           8Mi             
open-match-prometheus-node-exporter-qj2xd                   2m           8Mi             
open-match-prometheus-node-exporter-qwxnk                   0m           8Mi             
open-match-prometheus-node-exporter-ssm6n                   1m           8Mi             
open-match-prometheus-node-exporter-t6lxb                   2m           7Mi             
open-match-prometheus-node-exporter-wf8dn                   0m           8Mi             
open-match-prometheus-node-exporter-zngjl                   1m           7Mi             
open-match-prometheus-pushgateway-b97d7b99d-pjsss           1m           8Mi             
open-match-prometheus-server-554f785757-zdhdb               14m          273Mi        
```
